flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -256,9 +256,11 @@ public ExecutionConfig getExecutionConfig() {
 		DataStream<OUT> returnStream = this.copy();
 
 		for (DataStream<OUT> stream : streams) {
-			for (DataStream<OUT> ds : stream.unionedStreams) {
-				validateUnion(ds.getId());
-				returnStream.unionedStreams.add(ds.copy());
+			if (stream != null) {
```

Comment:
Right there will be an error, but the error is helpful because it indicates a problem with the user program. 
If the union is silently ignored, the program might behave differently from what the user expects and produce invalid results. 
I would argue, that a `union(null)` is never intended and should be be brought to the user's attention. Also, this error might reveal more problems of the user program.